File: python/render_sdk/experimental/experimental.py
"""Experimental Service

This module provides the ExperimentalService class that exposes experimental features
from the Render SDK.
"""

from typing import TYPE_CHECKING

from render_sdk.experimental.object.client import ObjectClient

if TYPE_CHECKING:
    from render_sdk.client.client import Client


class StorageService:
    """Storage Service

    Provides access to experimental storage features.

    Example:
        ```python
        # Access object storage
        await client.experimental.storage.objects.put(
            owner_id="tea-xxxxx",
            region="oregon",
            key="file.png",
            data=b"content"
        )
        ```
    """

    def __init__(self, client: "Client"):
        """Initialize the storage service.

        Args:
            client: The Render client instance
        """
        self.objects = ObjectClient(client.internal)


class ExperimentalService:
    """Experimental Service

    Provides access to experimental Render SDK features.

    Features in this namespace may change or be removed without a migration plan.
    When a feature stabilizes, it will be promoted to the main SDK namespace.

    Example:
        ```python
        from render_sdk.client import Client

        client = Client()

        # Access experimental object storage
        await client.experimental.storage.objects.put(
            owner_id="tea-xxxxx",
            region="oregon",
            key="file.png",
            data=b"content"
        )
        ```
    """

    def __init__(self, client: "Client"):
        """Initialize the experimental service.

        Args:
            client: The Render client instance
        """
        self.storage = StorageService(client)

File: typescript/src/experimental/experimental.ts
import type { Client } from "openapi-fetch";
import type { paths } from "../generated/schema.js";
import { ObjectClient } from "./object/client.js";

/**
 * StorageClient provides access to experimental storage features
 *
 * @example
 * ```typescript
 * // Access object storage
 * await render.experimental.storage.objects.put({
 *   ownerId: "tea-xxxxx",
 *   region: "oregon",
 *   key: "file.png",
 *   data: buffer
 * });
 * ```
 */
export class StorageClient {
  /** Object storage client for managing binary objects */
  public readonly objects: ObjectClient;

  constructor(apiClient: Client<paths>, defaultOwnerId?: string, defaultRegion?: string) {
    this.objects = new ObjectClient(apiClient, defaultOwnerId, defaultRegion);
  }
}

/**
 * ExperimentalClient provides access to experimental Render SDK features
 *
 * Features in this namespace may change or be removed without a migration plan.
 * When a feature stabilizes, it will be promoted to the main SDK namespace.
 *
 * @example
 * ```typescript
 * import { Render } from '@renderinc/sdk';
 *
 * const render = new Render();
 *
 * // Access experimental object storage
 * await render.experimental.storage.objects.put({
 *   ownerId: "tea-xxxxx",
 *   region: "oregon",
 *   key: "file.png",
 *   data: buffer
 * });
 * ```
 */
export class ExperimentalClient {
  /** Storage client for managing storage features */
  public readonly storage: StorageClient;

  constructor(apiClient: Client<paths>, defaultOwnerId?: string, defaultRegion?: string) {
    this.storage = new StorageClient(apiClient, defaultOwnerId, defaultRegion);
  }
}
